### PR TITLE
docs: add api proposal doc

### PR DESCRIPTION
**Pull request by junminahn**

### CODE REVIEW

File: docs/api-endpoints.md [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+- Responses
+
+<table style="margin-left: 2em;">
+<tr><td>Status Code</td><td>Response</td></tr>
+<tr>
+<td>200</td>
+<td>
+
+```json
+// the deleted role name
```
Comment (thegentlemanphysicist) [inline docs/api-endpoints.md:158]: is this supposed to be JSON

File: docs/api-endpoints.md [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+- Responses
+
+<table style="margin-left: 2em;">
+<tr><td>Status Code</td><td>Response</td></tr>
+<tr>
+<td>200</td>
+<td>
+
+```json
+// all roles for the user after the changes
```
Comment (thegentlemanphysicist) [inline docs/api-endpoints.md:280]: same as above

File: docs/api-endpoints.md (hunk 1 above)
Comment (junminahn) [reply docs/api-endpoints.md]: I moved the comment out of the json block.

File: docs/api-endpoints.md (hunk 2 above)
Comment (junminahn) [reply docs/api-endpoints.md]: I moved the comment out of the json block.

Comment (thegentlemanphysicist) [reply docs/api-endpoints.md]: thank you

**Pull request merged**